assets/js/base/components/product-grid-item/product-button/index.js

```diff
@@ -42,33 +33,47 @@ class ProductButton extends Component {
 				quantity: 1,
 			},
 			cache: 'no-store',
-		} ).then( ( response ) => {
-			const newQuantity = response.quantity;
+		} )
+			.then( ( response ) => {
+				const newQuantity = response.quantity;
+
+				this.setState( {
+					addedToCart: true,
+					addingToCart: false,
+					cartQuantity: newQuantity,
+				} );
+			} )
+			.catch( ( response ) => {
+				if ( response.code ) {
+					return ( document.location.href = addQueryArgs(
+						product.permalink,
+						{ wc_error: response.message }
+					) );
+				}
 
-			this.setState( {
-				buttonText: sprintf( __( '%d in cart', 'woo-gutenberg-products-block' ), newQuantity ),
-				addingToCart: false,
-				addedToCart: true,
+				document.location.href = product.permalink;
 			} );
-		} ).catch( ( response ) => {
-			if ( response.code ) {
-				return document.location.href = addQueryArgs( product.permalink, { wc_error: response.message } );
-			}
+	};
 
-			document.location.href = product.permalink;
-		} );
-	}
+	getButtonText = () => {
```

Comment:
Like it, but just noting this may change after design from @jwold - the X in cart text was something I added.